src/execution_plan/optimizations/apply_join.c

```diff
@@ -146,24 +89,28 @@ static void _reduce_cp_to_hashjoin(ExecutionPlan *plan, OpBase *cp) {
 
 		/* Make sure LHS of the filter is resolved by a stream. */
 		AR_ExpNode *lhs = f->pred.lhs;
-		uint lhs_resolving_stream = _relate_exp_to_stream(lhs, stream_entities, stream_count);
+		uint lhs_resolving_stream = OptimizeUtils_RelateExpToStream(lhs, stream_entities, stream_count);
 		if(lhs_resolving_stream == NOT_RESOLVED) continue;
 		/* Make sure RHS of the filter is resolved by a stream. */
 		AR_ExpNode *rhs = f->pred.rhs;
-		uint rhs_resolving_stream = _relate_exp_to_stream(rhs, stream_entities, stream_count);
+		uint rhs_resolving_stream = OptimizeUtils_RelateExpToStream(rhs, stream_entities, stream_count);
 		if(rhs_resolving_stream == NOT_RESOLVED) continue;
 
-		// This stream is solved by a single cartesian product child and needs to be propogate later.
+		/* Fix for issue #869 https://github.com/RedisGraph/RedisGraph/issues/869.
```

Comment:
Please remove this long explanation, simply explain the situation, no need to described a resolved bug.

Reply:
removed